From d7f58eeab1af87f1ade120186d250b86afdeb546 Mon Sep 17 00:00:00 2001
From: Ethan Cotterell <ethancotterell@gmail.com>
Date: Mon, 7 Jun 2021 12:29:51 +0100
Subject: [PATCH 04/11] argp-namefrob: add aliases to LIBC functions

---
 gnulib/gllib/argp-namefrob.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gnulib/gllib/argp-namefrob.h b/gnulib/gllib/argp-namefrob.h
index 022b880..ad8acf1 100644
--- a/gnulib/gllib/argp-namefrob.h
+++ b/gnulib/gllib/argp-namefrob.h
@@ -75,6 +75,8 @@
 #undef __argp_fmtstream_wmargin
 #define __argp_fmtstream_wmargin argp_fmtstream_wmargin
 
+#endif
+
 /* normal libc functions we call */
 #undef __flockfile
 #define __flockfile flockfile
@@ -95,6 +97,8 @@
 #undef __vsnprintf
 #define __vsnprintf vsnprintf
 
+#if !_LIBC
+
 #if defined(HAVE_DECL_CLEARERR_UNLOCKED) && !HAVE_DECL_CLEARERR_UNLOCKED
 # define clearerr_unlocked(x) clearerr (x)
 #endif
-- 
2.25.1

