From dc846f857fbcc18c040104c398bac064ff9d0678 Mon Sep 17 00:00:00 2001
From: Ethan Cotterell <ethancotterell@gmail.com>
Date: Mon, 7 Jun 2021 16:27:51 +0100
Subject: [PATCH 10/11] getopt1: remove duplicate global vars and LIBC
 dependent functions

---
 gnulib/gllib/getopt1.c | 40 +++++++++++++++++++++-------------------
 1 file changed, 21 insertions(+), 19 deletions(-)

diff --git a/gnulib/gllib/getopt1.c b/gnulib/gllib/getopt1.c
index 0e398e0..5466db6 100644
--- a/gnulib/gllib/getopt1.c
+++ b/gnulib/gllib/getopt1.c
@@ -16,21 +16,22 @@
    You should have received a copy of the GNU Lesser General Public
    License along with the GNU C Library; if not, see
    <https://www.gnu.org/licenses/>.  */
-
-#ifndef _LIBC
+
+
+// #ifndef _LIBC
 # include <config.h>
-#endif
+// #endif
 
 #include "getopt.h"
 #include "getopt_int.h"
 
-int
-getopt_long (int argc, char *__getopt_argv_const *argv, const char *options,
-	     const struct option *long_options, int *opt_index)
-{
-  return _getopt_internal (argc, (char **) argv, options, long_options,
-			   opt_index, 0, 0);
-}
+// int
+// getopt_long (int argc, char *__getopt_argv_const *argv, const char *options,
+// 	     const struct option *long_options, int *opt_index)
+// {
+//   return _getopt_internal (argc, (char **) argv, options, long_options,
+// 			   opt_index, 0, 0);
+// }
 
 int
 _getopt_long_r (int argc, char **argv, const char *options,
@@ -46,14 +47,14 @@ _getopt_long_r (int argc, char **argv, const char *options,
    but does match a short option, it is parsed as a short option
    instead.  */
 
-int
-getopt_long_only (int argc, char *__getopt_argv_const *argv,
-		  const char *options,
-		  const struct option *long_options, int *opt_index)
-{
-  return _getopt_internal (argc, (char **) argv, options, long_options,
-			   opt_index, 1, 0);
-}
+// int
+// getopt_long_only (int argc, char *__getopt_argv_const *argv,
+// 		  const char *options,
+// 		  const struct option *long_options, int *opt_index)
+// {
+//   return _getopt_internal (argc, (char **) argv, options, long_options,
+// 			   opt_index, 1, 0);
+// }
 
 int
 _getopt_long_only_r (int argc, char **argv, const char *options,
@@ -64,7 +65,8 @@ _getopt_long_only_r (int argc, char **argv, const char *options,
 			     1, d, 0);
 }
 
-
+
+
 #ifdef TEST
 
 #include <stdio.h>
-- 
2.25.1

